nautobot/dcim/forms.py

```diff
@@ -2216,6 +2228,36 @@ def __init__(self, *args, **kwrags):
         # Disable position because only setting null value is required
         self.fields["position"].disabled = True
 
+    def save(self, pk_list):
+        # Call the parent method first to handle the standard field updates
+        result = super().save(pk_list)
+
+        # Process clusters
+        devices = Device.objects.filter(pk__in=pk_list)
+        add_clusters = self.cleaned_data.get("add_clusters")
+        remove_clusters = self.cleaned_data.get("remove_clusters")
+        remove_all_clusters = self.cleaned_data.get("remove_all_clusters")
+
+        # TODO: This isn't working for unclear reasons.
+        if remove_all_clusters:
+            DeviceClusterAssignment.objects.filter(device__in=devices).delete()
+            return result
+
+        if add_clusters:
+            missing_assignments = []
+            for device in devices:
+                for cluster in add_clusters:
+                    if not DeviceClusterAssignment.objects.filter(device=device, cluster=cluster).exists():
+                        missing_assignments.append(DeviceClusterAssignment(device=device, cluster=cluster))
+
+            if missing_assignments:
+                DeviceClusterAssignment.objects.bulk_create(missing_assignments)
+
+        if remove_clusters:
+            DeviceClusterAssignment.objects.filter(device__in=devices, cluster__in=remove_clusters).delete()
+
+        return result
```

Comment:
`add_` and `remove_` should be already handled generically in bulk edit form and the bulk edit job. Was that not the case for this one?

Reply:
nope, my own ignorance. removed.

also removed the remove from all clusters form option, unless you think it belongs there